src/pages/manage/productions/index.js

```diff
@@ -45,22 +45,25 @@ const Index = () => {
     start: currentPageProductions - 1,
     limit: productionsPerPage,
   });
+  const rawProductions = productionsData?.member ?? [];
+
+  useEffect(() => {
+    if (rawProductions.length === 0) return;
+    // make the first production active when new data comes in from the apiCall
+    setActiveProductionId(rawProductions[0].production_id);
+  }, [rawProductions]);
 
   const productions = useMemo(() => {
-    const rawProductions = productionsData?.member ?? [];
     return rawProductions.map((production) => ({
       ...production,
       id: production.production_id,
       active: production.production_id === activeProductionId,
     }));
-  }, [productionsData, activeProductionId]);
-
-  useEffect(() => setActiveProductionId(productions?.[0]?.id), [productions]);
+  }, [activeProductionId]);
```

Comment:
I think we also need to pass rawProductions here then?

Reply:
I don't think so because setActiveProductionId gets called when rawProductions updates